cmd/ipi-deprovision/ipi-deprovision.sh

```diff
@@ -1,17 +1,44 @@
 #!/bin/sh
 
+set -o errexit
+set -o nounset
+set -o pipefail
+
+if [ -z "${AWS_SHARED_CREDENTIALS_FILE}" ]; then echo "ENV VAR AWS_SHARED_CREDENTIALS_FILE is not set"; exit 1; fi
+
 ### expirationDate is always with timezone +00:00, eg, "2019-07-23T22:35+0000"
-MY_DATE=$(TZ=":Africa/Abidjan" date '+%Y-%m-%d')
-declare -a regions=("us-east-1" "us-east-2" "us-west-1")
+my_date=$(TZ=":Africa/Abidjan" date '+%Y-%m-%d')
+
+if [[ -n "${IPI_DEPROVISION_DATE_ARG-}" ]]; then
+  ### eg, export IPI_DEPROVISION_DATE_ARG="22 hours ago"
+  echo "take env. var.: IPI_DEPROVISION_DATE_ARG: ${IPI_DEPROVISION_DATE_ARG}"
+  my_date="$(TZ=":Africa/Abidjan" date --date="${IPI_DEPROVISION_DATE_ARG}" '+%Y-%m-%dT%H:%M+0000')"
+fi
+
+echo "my_date: ${my_date}"
+
+### James: CI only runs in us-east-1.
+declare -a regions=("us-east-1")
+### for more than one region:
+# declare -a regions=("us-east-1" "us-east-2" "us-west-1")
+if [[ -n "${IPI_DEPROVISION_REGIONS-}" ]]; then
+  ### eg, export IPI_DEPROVISION_REGIONS="us-east-1;us-west-1"
+  echo "take regions from env. var.: IPI_DEPROVISION_REGIONS: ${IPI_DEPROVISION_REGIONS}"
+  IFS=';' read -r -a regions <<< "$IPI_DEPROVISION_REGIONS"
+fi
+
+echo "regions: ${regions}"
 
 handle_cluster () {
   local cluster_name;
   cluster_name=$1
   echo "handling cluster: ${cluster_name} ..."
+  ### TODO: will test hiveutil later
+  #./bin/hiveutil aws-tag-deprovision "${cluster_name}=owned"
 }
 
 for r in "${regions[@]}"
 do
   echo "doing region ${r} ..."
-  aws ec2 describe-vpcs --output json --region "${r}" | jq --arg date "${MY_DATE}" -r -S '.Vpcs[] | select (.Tags[]? | (.Key == "expirationDate" and .Value < $date)) | .Tags[] | select (.Value == "owned") | .Key' | while read line; do handle_cluster ${line}; done
+  aws ec2 describe-vpcs --output json --region "${r}" | jq --arg date "${my_date}" -r -S '.Vpcs[] | select (.Tags[]? | (.Key == "expirationDate" and .Value < $date)) | .Tags[] | select (.Value == "owned") | .Key' | while read line; do handle_cluster ${line}; done
```

Comment:
nit: piping into `while read line` is a bit esoteric, we could:


```sh
for cluster in $( aws ec2 describe... ); do
    handle_cluster "${cluster}"
done
```

(and with that formatting we hardly need the `handle_cluster` function since inlining it is simple

Reply:
Done.